controllers/promise_controller.go

```diff
@@ -174,6 +188,54 @@ func (r *PromiseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	return ctrl.Result{}, nil
 }
 
+func (r *PromiseReconciler) reconcileDependencies(o opts, promise *v1alpha1.Promise, configurePipeline []v1alpha1.Pipeline) (*ctrl.Result, error) {
+	if len(promise.Spec.Workflows.Promise.Configure) == 0 {
+		o.logger.Info("Promise does not contain workflows.promise.configure, applying dependencies directly")
+		if err := r.applyWorkResourceForDependencies(o, promise); err != nil {
+			o.logger.Error(err, "Error creating Works")
+			return nil, err
+		}
+	} else {
+		if doesNotContainFinalizer(promise, workflowsFinalizer) {
+			result, err := addFinalizers(o, promise, []string{workflowsFinalizer})
+			return &result, err
+		}
+
+		o.logger.Info("Promise contains workflows.promise.configure, reconciling workflows")
+		objMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&promise)
+		if err != nil {
+			return nil, err
+		}
+		unstructuredPromise := &unstructured.Unstructured{Object: objMap}
+		pipelineResources, err := pipeline.NewConfigurePromise(
+			unstructuredPromise,
+			configurePipeline,
+			promise.GetName(),
+			promise.Spec.DestinationSelectors,
+			o.logger,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		jobOpts := jobOpts{
+			opts:              o,
+			obj:               unstructuredPromise,
+			pipelineLabels:    pipeline.LabelsForConfigurePromise(promise.GetName()),
+			pipelineResources: pipelineResources,
+		}
+		requeue, err := ensurePipelineIsReconciled(jobOpts)
+		if err != nil {
+			return nil, err
+		}
+
+		if requeue != nil {
+			return requeue, nil
+		}
+	}
+	return nil, nil
```

Comment:
it looks like you can just return `requeue` and `err`?

```suggestion
	return ensurePipelineIsReconciled(jobOpts)
```